app/Listeners/ProspectActivityListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Listeners;
+
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use App\Repositories\ProspectActivityRepository;
+use Carbon\Carbon;
+
+class ProspectActivityListener
+{
+    /**
+     * Create the event listener.
+     *
+     * @return void
+     */
+    private $prospectActivityRepo;
+    public function __construct(ProspectActivityRepository $prospectActivityRepository)
+    {
+        //
+        $this->prospectActivityRepo = $prospectActivityRepository;
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param  object  $event
+     * @return void
+     */
+    public function handle($event)
+    {
+        $this->prospectActivityRepo->store(
+            [
+                'contact_customer_id' => $event->data->customer_id,
+                'user_id' => auth()->id() ?? null,
+                'type' => strtolower(class_basename(get_class($event->data))),
+                'prospect_activity_type' => get_class($event->data),
```

Comment:
Do we still this class?

Reply:
yes we are using it

Reply:
why is laravel not resolving it?

Reply:
It is because of how I am saving the data.

Laravel needs the model instance like 
$feedback->save()
then
$feedback->prospectActivti->create([data......])

But after passing the data via even model, it has been converted into collection, which means the create method is not accessible on the $event->data object.